Sweetener.Logging/Loggers/Logger.Format.cs

```diff
@@ -0,0 +1,807 @@
+
+            if (MinimumLevel <= LogLevel.Debug)
+                Log(LogLevel.Debug, string.Format(FormatProvider, format, arg0, arg1));
+        }
+
+        /// <summary>
+        /// Request to log the text representation of the specified objects at the
+        /// <see cref="LogLevel.Debug"/> level using the specified format information.
+        /// </summary>
+        /// <remarks>
+        /// The log request will only be fulfilled if the <see cref="MinimumLevel"/> is less
+        /// than or equal to <see cref="LogLevel.Debug"/>.
+        /// </remarks>
+        /// <param name="format">A composite format string.</param>
+        /// <param name="arg0">The first object to write using <paramref name="format"/>.</param>
+        /// <param name="arg1">The second object to write using <paramref name="format"/>.</param>
+        /// <param name="arg2">The third object to write using <paramref name="format"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="format"/> is <c>null</c>.</exception>
+        /// <exception cref="FormatException">The format specification in <paramref name="format"/> is invalid.</exception>
+        /// <exception cref="ObjectDisposedException">The logger is disposed.</exception>
+        public void Debug(string format, object arg0, object arg1, object arg2)
+        {
+            ThrowIfDisposed();
+
+            if (format == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.format);
+
+            if (MinimumLevel <= LogLevel.Debug)
+                Log(LogLevel.Debug, string.Format(FormatProvider, format, arg0, arg1, arg2));
+        }
+
+        /// <summary>
+        /// Request to log the text representation of the specified objects at the
+        /// <see cref="LogLevel.Debug"/> level using the specified format information.
+        /// </summary>
+        /// <remarks>
+        /// The log request will only be fulfilled if the <see cref="MinimumLevel"/> is less
+        /// than or equal to <see cref="LogLevel.Debug"/>.
+        /// </remarks>
+        /// <param name="format">A composite format string.</param>
+        /// <param name="arg0">The first object to write using <paramref name="format"/>.</param>
+        /// <param name="arg1">The second object to write using <paramref name="format"/>.</param>
+        /// <param name="arg2">The third object to write using <paramref name="format"/>.</param>
+        /// <param name="arg3">The fourth object to write using <paramref name="format"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="format"/> is <c>null</c>.</exception>
+        /// <exception cref="FormatException">The format specification in <paramref name="format"/> is invalid.</exception>
+        /// <exception cref="ObjectDisposedException">The logger is disposed.</exception>
+        public void Debug(string format, object arg0, object arg1, object arg2, object arg3)
```

Comment:
What is the value of this and similar methods if we have varargs?

Reply:
This mirrors a lot of the .NET APIs like string.Format and Console.WriteLine. The advantage with these overloads is that we avoid the array allocation if you only need a few arguments.